python/lsst/obs/subaru/gen3.py

```diff
@@ -95,42 +95,37 @@ def extract(self, gen2id, skyMap, skyMapName):
         return physical
 
 
-class HscPhysicalFilterKeyHandler(KeyHandler):
-    """KeyHandler for HSC filter keys that should be mapped to PhysicalFilters.
-    """
-
-    __slots__ = ()
-
-    def __init__(self):
-        super().__init__("physical_filter", "PhysicalFilter")
-
-    def translate(self, gen2id, gen3id, skyMap, skyMapName):
-        super().translate(gen2id, gen3id, skyMap, skyMapName)
-        # also add camera
-        gen3id["camera"] = "HSC"
-
-    def extract(self, gen2id, skyMap, skyMapName):
-        return gen2id["filter"]
-
-
 # Add camera to Gen3 data ID if Gen2 contains "visit" or "ccd".
 # (Both rules will match, so we'll actually set camera in the same dict twice).
 Translator.addRule(ConstantKeyHandler("camera", "Camera", "HSC"),
                    camera="HSC", gen2keys=("visit",), consume=False)
 Translator.addRule(ConstantKeyHandler("camera", "Camera", "HSC"),
                    camera="HSC", gen2keys=("ccd",), consume=False)
 
-# Copy Gen2 'visit' to Gen3 'exposure' for raw only.
+# Copy Gen2 'visit' to Gen3 'exposure' for raw only.  Also consume filter,
+# since that's implied by 'exposure' in Gen3.
 Translator.addRule(CopyKeyHandler("exposure", "Exposure", "visit"),
-                   camera="HSC", datasetTypeName="raw", gen2keys=("visit",))
+                   camera="HSC", datasetTypeName="raw", gen2keys=("visit",),
+                   consume=("visit", "filter"))
```

Comment:
What happens with `filter`? Shouldn't this go to `abstract_filter` or `physical_filter`?

Reply:
It turns out we don't have any datasets that use both Visit and AbstractFilter, and Visit + PhysicalFilter is redundant, because a Visit knows its own PhysicalFilter (or, in other words, Visit's dependency on PhysicalFilter is optional).  So stripping `filter` whenever we see `visit` keeps us from adding an incorrect "required" dependency on PhysicalFilter.

We do translate `filter` to `abstract_filter` when its paired with `tract` (via a later rule), and there's at least one DatasetType-specific rule for the one case where we have a PhysicalFilter that isn't from a Visit or Exposure.